modules/azure/compute.go

```diff
@@ -8,97 +8,444 @@ import (
 	}
 
-	// Get the details of the target virtual machine
-	vm, err := vmClient.Get(context.Background(), resGroupName, vmName, compute.InstanceView)
+	return string(*vm.OsProfile.AdminUsername), nil
+}
+
+// GetVirtualMachineSize gets the Size Type of the specified Azure Virtual Machine.
+// This function would fail the test if there is an error.
+func GetVirtualMachineSize(t testing.TestingT, vmName string, resGroupName string, subscriptionID string) compute.VirtualMachineSizeTypes {
+	size, err := GetVirtualMachineSizeE(vmName, resGroupName, subscriptionID)
+	require.NoError(t, err)
+
+	return size
+}
+
+// GetVirtualMachineSizeE gets the Size Type of the specified Azure Virtual Machine.
+func GetVirtualMachineSizeE(vmName string, resGroupName string, subscriptionID string) (compute.VirtualMachineSizeTypes, error) {
+	// Get VM Object
+	vm, err := GetVirtualMachineE(vmName, resGroupName, subscriptionID)
 	if err != nil {
 		return "", err
 	}
 
 	return vm.VirtualMachineProperties.HardwareProfile.VMSize, nil
 }
 
-// GetTagsForVirtualMachine gets the tags of the given Virtual Machine as a map.
+// GetVirtualMachineTags gets the Tags of the specified Virtual Machine as a map.
 // This function would fail the test if there is an error.
-func GetTagsForVirtualMachine(t testing.TestingT, vmName string, resGroupName string, subscriptionID string) map[string]string {
-	tags, err := GetTagsForVirtualMachineE(t, vmName, resGroupName, subscriptionID)
+func GetVirtualMachineTags(t testing.TestingT, vmName string, resGroupName string, subscriptionID string) map[string]string {
+	tags, err := GetVirtualMachineTagsE(vmName, resGroupName, subscriptionID)
 	require.NoError(t, err)
 
 	return tags
 }
 
-// GetTagsForVirtualMachineE gets the tags of the given Virtual Machine as a map
-func GetTagsForVirtualMachineE(t testing.TestingT, vmName string, resGroupName string, subscriptionID string) (map[string]string, error) {
+// GetVirtualMachineTagsE gets the Tags of the specified Virtual Machine as a map.
+func GetVirtualMachineTagsE(vmName string, resGroupName string, subscriptionID string) (map[string]string, error) {
 	// Setup a blank map to populate and return
-	tags := make(map[string]string)
+	var tags map[string]string
+
+	// Get VM Object
+	vm, err := GetVirtualMachineE(vmName, resGroupName, subscriptionID)
+	if err != nil {
+		return nil, err
+	}
+
+	// Range through existing tags and populate above map accordingly
+	for k, v := range vm.Tags {
+		tags[k] = *v
+	}
+
+	return tags, nil
+}
+
+// ***************************************************** //
+// Get multiple Virtual Machines from a Resource Group
+// ***************************************************** //
+
+// GetResourceGroupVirtualMachines gets a list of all Virtual Machine names in the specified Resource Group.
+// This function would fail the test if there is an error.
+func GetResourceGroupVirtualMachines(t testing.TestingT, resGroupName string, subscriptionID string) []string {
```

Comment:
should function name be ListVirtualMachinesForResourceGroup?

Reply:
Updated